src/completions.rs

```diff
@@ -15,54 +15,76 @@ pub struct Completions<T: LLMModel> {
     model: T,
     //For prompt & response
     max_tokens: usize,
-    temperature: u32,
+    temperature: f32,
     input_json: Option<String>,
     debug: bool,
     function_call: bool,
     api_key: String,
 }
 
 impl<T: LLMModel> Completions<T> {
-    ///
+    /// Constructor for the Completions API
     pub fn new(
         model: T,
         api_key: &str,
         max_tokens: Option<usize>,
         temperature: Option<u32>,
     ) -> Self {
+        let temperature = temperature
+            .map(|temp| model.get_normalized_temperature(temp))
+            .unwrap_or(model.get_default_temperature());
         Completions {
             //If no max tokens limit is provided we default to max allowed for the model
             max_tokens: max_tokens.unwrap_or_else(|| model.default_max_tokens()),
             function_call: model.function_call_default(),
             model,
-            temperature: temperature.unwrap_or(0u32), //Low number makes the output less random and more deterministic
+            temperature,
             input_json: None,
             debug: false,
             api_key: api_key.to_string(),
         }
     }
 
-    /*
-     * This function turns on debug mode which will info! the prompt to log when executing it.
-     */
+    ///
+    /// This function turns on debug mode which will info! the prompt to log when executing it.
+    ///
     pub fn debug(mut self) -> Self {
         self.debug = true;
         self
     }
 
-    /*
-     * This function turns on/off function calling mode when interacting with OpenAI API.
-     */
+    ///
+    /// This function turns on/off function calling mode when interacting with OpenAI API.
+    ///
     pub fn function_calling(mut self, function_call: bool) -> Self {
         self.function_call = function_call;
         self
     }
 
-    /*
-     * This method can be used to provide values that will be used as context for the prompt.
-     * Using this function you can provide multiple input values by calling it multiple times. New values will be appended with the category name
-     * It accepts any instance that implements the Serialize trait.
-     */
+    ///
+    /// This method can be used to define the model temperature used by the Assistant
+    /// This method accepts % target of the acceptable range for the model
+    ///
+    pub fn temperature(mut self, temp_target: u32) -> Self {
```

Comment:
Added the same methods for `Completions` as for `Assistants` to keep the interfaces consistent.